src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/scanner/ReaderStringTest.kt

```diff
@@ -0,0 +1,110 @@
+
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.assertions.withClue
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.string.shouldContain
+import it.krzeminski.snakeyaml.engine.kmp.api.LoadSettings
+import it.krzeminski.snakeyaml.engine.kmp.exceptions.ReaderException
+
+class ReaderStringTest : FunSpec({
+    test("check printable") {
+        val reader = StreamReader(LoadSettings.builder().build(), "test")
+        reader.peek(4) shouldBe 0
+        StreamReader.isPrintable("test") shouldBe true
+    }
+
+    test("check non-printable") {
+        StreamReader.isPrintable("test\u0005 fail") shouldBe false
+        shouldThrow<ReaderException> {
+            val reader = StreamReader(LoadSettings.builder().build(), "test\u0005 fail")
+            while (reader.peek() != 0) {
+                reader.forward()
+            }
+        }.also {
+            it.toString() shouldBe "unacceptable code point '\u0005' (0x5) special characters are not allowed\nin \"reader\", position 4"
+        }
+    }
+
+    test("check all") {
+        var highSurrogatesCounter = 0
+        var lowSurrogatesCounter = 0
+        for (i in 0..<256 * 256) {
+            if (i.toChar().isHighSurrogate()) {
+                highSurrogatesCounter++
+            } else if (i.toChar().isLowSurrogate()) {
+                lowSurrogatesCounter++
+            } else {
+                val str = CharArray(1) { i.toChar() }.concatToString()
+                val regularExpressionResult = StreamReader.isPrintable(str)
+
+                var charsArrayResult = true
+                try {
+                    StreamReader(LoadSettings.builder().build(), str).peek()
+                } catch (e: Exception) {
+                    (
+                        (e.message?.startsWith("unacceptable character") ?: false)
+                            || e.message == "special characters are not allowed"
+                    ) shouldBe true
+                    charsArrayResult = false
+                }
+                withClue("Failed for #$i") {
+                    regularExpressionResult shouldBe charsArrayResult
+                }
+            }
+        }
+        // https://en.wikipedia.org/wiki/Universal_Character_Set_characters
+        withClue("There are 1024 low surrogates (D800–DBFF)") {
+            highSurrogatesCounter shouldBe 1024
+        }
+        withClue("There are 1024 high surrogates (DC00–DFFF)") {
+            lowSurrogatesCounter shouldBe 1024
+        }
+    }
+
+    test("high surrogates alone") {
+        for (i in 0xD800..0xDBFF) {
+            val str = CharArray(1) { i.toChar() }.concatToString()
+            val reader = StreamReader(LoadSettings.builder().build(), str)
+            reader.peek() shouldBe '?'.code
+        }
+    }
+
+    test("low surrogates alone") {
+        for (i in 0xDC00..0xDFFF) {
+            val str = CharArray(1) { i.toChar() }.concatToString()
```

Comment:
suggestion: `CharArray(1) {}` can be replaced with `charArrayOf(...)`

Reply:
I just noticed that you have a requirement to have all conversations resolved - please, feel free to resolve any comments that are not marked as issue😊

Reply:
Thanks, let me apply this!

Reply:
https://github.com/krzema12/snakeyaml-engine-kmp/pull/314/commits/5177cda7d3bb271ea93327b5c36b0da96bfca173